File: app/src/main/java/com/kuanquan/universalcomponents/SearchActivity.java
package com.kuanquan.universalcomponents;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;
import com.base.library.widget.ClearEditText;
import com.base.library.widget.SearchEditText;

/**
 * 搜索
 */
public class SearchActivity extends AppCompatActivity implements SearchEditText.OnSearchClickListener{

    private SearchEditText searchEditText;
    private ClearEditText mClearEditText;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);

        searchEditText = (SearchEditText) findViewById(R.id.searchEditText);

        //搜索事件
        searchEditText.setOnSearchClickListener(this);

        mClearEditText = findViewById(R.id.et_earch);
    }

    /**
     * 搜索事件
     */
    @Override
    public void onSearchClick(View view, String keyword) {
        if (!TextUtils.isEmpty(keyword)) {
            //在这里处理逻辑
            Toast.makeText(this, keyword, Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/kuanquan/universalcomponents/SearchActivity.kt
package com.kuanquan.universalcomponents

import android.support.v7.widget.LinearLayoutManager
import android.text.TextUtils
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.TextView
import com.base.library.base.BaseViewModelActivity
import com.base.library.utils.KeyBoard
import com.base.library.utils.LogUtil
import com.kuanquan.universalcomponents.adapter.SearchAdapter
import com.kuanquan.universalcomponents.viewmodel.SearchViewModel
import kotlinx.android.synthetic.main.activity_search.*

class SearchActivity : BaseViewModelActivity<SearchViewModel>() {

    lateinit var adapter: SearchAdapter

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.search_cancel -> finish()
        }
    }

    override fun createViewModel(): SearchViewModel {
        return createViewModel(this, SearchViewModel::class.java)
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_search
    }

    override fun initView() {
        super.initView()
        addOnClickListeners(this, R.id.search_cancel)
        search_recycler_view.layoutManager = LinearLayoutManager(this)

        val content = et_earch.text.toString().trimMargin()

        et_earch.setOnEditorActionListener(object : TextView.OnEditorActionListener {
            override fun onEditorAction(v: TextView, actionId: Int, event: KeyEvent): Boolean {
                if (actionId == EditorInfo.IME_ACTION_SEARCH) {  //搜索按键 action
                    KeyBoard.hintKbTwo(this@SearchActivity,this@SearchActivity.window)
                    if (TextUtils.isEmpty(content)) {
                        return true
                    }
                    LogUtil.e("开始搜索")
                    return true
                }
                return false
            }
        })
    }

    override fun initData() {
        adapter = SearchAdapter(mViewModel.default())
        search_recycler_view.adapter = adapter
    }

    override fun isBindEventBusHere(): Boolean {
        return false
    }

    override fun dataObserver() {
    }
}